File: Arrays/BinarySearch.py
def binary_search(arr, target):
    low = 0
    high = len(arr) - 1
    
    while (low <=high):
        mid = (low + high) // 2
        
        if arr[mid] == target:
            return mid
        elif target > arr[mid]:
            low = mid + 1
        else:
            high = mid - 1
            
    return -1

# Example usage
arr = [1, 2, 3, 4, 5, 6, 7, 8, 9]
target = 5
result = binary_search(arr, target)
print(f"Element {target} is at index {result}")  
            
            

File: Arrays/BinarySearch.java
package Arrays;

public class BinarySearch {
    public static int binarySearch(int[] arr, int target){
        int low = 0; 
        int high = arr.length - 1;

        while(low <= high){
            int mid = (low + high) / 2;
            if (arr[mid] == target){
                return mid; 
           }
           else if(arr[mid] < target){
            low = mid + 1;
           }
            else{
                high = mid - 1;
            }
           }
           return -1;
        }

    public static void main(String[] args) {
        int[] arr = {1, 2, 3, 4, 5};
        int target = 4;
        int result = binarySearch(arr, target);
        if (result != -1) {
            System.out.println("Element found at index: " + result);
        } else {
            System.out.println("Element not found in the array.");
        }
    }
}
